app/models/concerns/course/video/submission/todo_concern.rb

```diff
@@ -17,10 +17,12 @@ def todo
   end
 
   def complete_todo
-    todo.update_attribute(:workflow_state, 'completed') unless todo.completed?
+    todo.update_column(:workflow_state, 'completed') unless todo.completed?
   end
 
+  # Skip callback if video is deleted as todo will be deleted.
   def restart_todo
-    todo.update_attribute(:workflow_state, 'not_started') unless todo.not_started?
+    return if video.acting_as.destroyed?
```

Comment:
try to use `video.destroying?` ? this is our custom callback to tell if an object is being destoryed. 

`destroyed?` is not reliable cause it only returns true when the object is already destroyed.